src/utils/MemoryLockWrapper.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include "utils/MemoryLockCore.h"
+
+namespace nebula {
+
+// RAII style to easily control the lock acquire / release
+template<class Key>
+class MemoryLockGuard {
+public:
+    MemoryLockGuard(MemoryLockCore<Key>* lock, const Key& key)
+        : MemoryLockGuard(lock, std::vector<Key>{key}) {}
+
+    explicit MemoryLockGuard(MemoryLockCore<Key>* lock, const std::vector<Key>& key)
```

Comment:
Great job. all are good for me.
A small suggestion, could we add a new constructor method as below?
`  explicit MemoryLockGuard(MemoryLockCore<Key>* lock, const std::set<Key>& key)`
A scene like this, Insert duplicate values in insert request, such as : 
`INSERT VERTEX t1(c1) VALUES "1":("1"), "1":("2"), 1:("3")`
the `key` should be {"1"},  not is {"1", "1", "1"}
Will be easier to use. WDYT?

Reply:
I add a constructor, which has a bool parameter 'dedup'.